### ISSUE TITLE
stack: streamingProcess: runInteractiveProcess: exec: does not exist

### ISSUE THREAD

**Issue by worm2fed** [haskell/haskell-language-server#1799]
I don't know what happened, but now I have following error:
`stack: streamingProcess: runInteractiveProcess: exec: does not exist (No such file or directory)`

But! In output (logs) I can't see any error:

Just for example:
```
2021-05-04 20:16:39.498227 [ThreadId 84] DEBUG hls:	LOOKUP UP PERSISTENT FOR: GetHieAst
2021-05-04 20:16:39.50861 [ThreadId 57] INFO hls:	finish: CodeAction (took 0.02s)
2021-05-04 20:16:39.513626 [ThreadId 57] INFO hls:	finish: importLens (took 0.03s)
2021-05-04 20:16:39.51376 [ThreadId 57] DEBUG hls:	LOOKUP UP PERSISTENT FOR: TypeCheck
2021-05-04 20:16:39.51384 [ThreadId 57] INFO hls:	finish: retrie (took 0.03s)
2021-05-04 20:16:39.516834 [ThreadId 84] DEBUG hls:	LOADING HIE FILE :NormalizedFilePath "/Users/worm2fed/workspace/serokell/edna/backend/src/Edna/Orphans.hs"
2021-05-04 20:16:39.526469 [ThreadId 84] DEBUG hls:	SUCCEEDED LOADING HIE FILE FOR:"/Users/worm2fed/.cache/ghcide/main-29ba7bf8fee79b993ce21c17cf9224089b04bc42/Edna/Orphans.hie"
2021-05-04 20:16:39.526658 [ThreadId 84] INFO hls:	finish: tactic (took 0.04s)
2021-05-04 20:16:39.722287 [ThreadId 96] INFO hls:	finish: codeLens.TypeCheck (took 0.00s)
2021-05-04 20:16:39.722534 [ThreadId 98] INFO hls:	finish: ModuleName.ghcSession (took 0.00s)
2021-05-04 20:16:39.722634 [ThreadId 100] INFO hls:	finish: codeLens.GetBindings (took 0.00s)
2021-05-04 20:16:39.722714 [ThreadId 102] INFO hls:	finish: codeLens.GetGlobalBindingTypeSigs (took 0.00s)
2021-05-04 20:16:39.723402 [ThreadId 92] DEBUG hls:	src/Ide/Plugin/Eval/CodeLens.hs:149:15 "fp" 
....
```

I use hls from ghcup:
```
✓  ghc   8.10.3         base-4.14.1.0             hls-powered
✔✔ ghc   8.10.4         recommended,base-4.14.1.0 hls-powered
✗  ghc   9.0.1          latest,base-4.15.0.0
✗  ghc   9.2.0.20210422 prerelease,base-4.16.0.0
✗  cabal 2.4.1.0
✗  cabal 3.0.0.0
✗  cabal 3.2.0.0
✔✔ cabal 3.4.0.0        latest,recommended
✔✔ hls   1.1.0          latest,recommended
✔✔ ghcup 0.1.14.1       latest,recommended
```

stack --version                                                                                                                                                                                                      
```
Version 2.5.1, Git revision d6ab861544918185236cf826cb2028abb266d6d5 x86_64 hpack-0.33.0
```

uname -a
```
Darwin tarvos.local 20.4.0 Darwin Kernel Version 20.4.0: Thu Apr 22 21:46:47 PDT 2021; root:xnu-7195.101.2~1/RELEASE_X86_64 x86_64
```

**Comment by jneira** [comment]
@worm2fed hi, thanks for your bug report, it seems hls wants to run stack but it cant find it in `$PATH`. Maybe are you trying to load a project with `stack.yaml` but without `cabal.project`? In that case hls assume the project has to be loaded with stack.

**Comment by worm2fed** [comment]
@jneira hi! 

here is my `echo $PATH` out:
```
...
/Users/worm2fed/.nix-profile/bin 
/Users/worm2fed/.cabal/bin 
/Users/worm2fed/.ghcup/bin
```

And `which stack`                                                                                                                                                                                                      
```
/Users/worm2fed/.local/bin/stack
```

Also, I have `cabal.project` and `hie.yaml` too!

But, hls works fine (almost, I have another issue). Yesterday I've installed nix, and after it hls in vscose doesn't work any more.

Today I've tried to disable nix in my `~/.stack/config.yaml` 
```
...
nix:
  enable: false
```
But it doesn't help (and as I can see here (https://docs.haskellstack.org/en/stable/nix_integration/#configuration) nix integration is switched off by default..

Maybe hls has something like `--no-nix`?
